File: app/models/review.rb
class Review < ApplicationRecord
  belongs_to :reviewable, polymorphic: true
  belongs_to :user
  validates :rating, presence: true, numericality: { only_integer: true, greater_than: 0, less_than_or_equal_to: 5 }
  validates :description, presence: true, length: { minimum: 50 }
  validates :user_id, uniqueness: { scope: [:reviewable_type, :reviewable_id], message: "can add only one review per resource" }
end

File: app/javascript/packs/api/review.js
import Http from './http';

export default {
  index(id, type){
    return Http.get('/reviews', {
      params: {
        id: id,
        type: type
      }
    });
  },
  create(id, type, description, rating){  
    return Http.post('/reviews', {
      review: {
        reviewable_id: id,
        reviewable_type: type.replace(/\b\w/g, l => l.toUpperCase()),
        description: description,
        rating: rating
      }
    });
  }
}